### DISCUSSION TITLE
Debian

### DISCUSSION THREAD

**Discussion by kainers1** [Freetz-NG/freetz-ng#667] [Allgemein]
Habe am Rande mitbekommen, dass es Probleme mit Debian gibt. Die Diskussion dazu ist leider zu. 

Welche sind das konkret @karstengit?

Kannst du mal deine .config posten mit der du Probleme hast?

Ich verwende seit eh und je (fritzbox 7170) Debian zum bauen und hatte nie schwerwiegende Probleme.
Habe erst gestern für #662 eine neue VM mit Debian 11.5 erstellt. 
Prerequisites habe ich 1:1 von Ubuntu 22 übernommen. Execstack musste ich aus Debian 10 nehmen, ist für 11 nicht verfügbar (warum auch immer). Sbin muss noch in den Path.

**Comment by aus-sua** [comment]
> ... dass es Probleme mit Debian gibt... Welche sind das konkret..

Wenn Du Zeit hast, Dir da 'mal die Mühe machst, und Dich durch seine vielen
eröffneten Issues und Discussions durchkämpfst, wirst Du sehen, daß es da schon
bei den Prerequisites gescheitert war, weil er sie einfach nicht gelesen und wohl
auch nicht verstanden hast...
Erste Fehlermeldungen war ja auch nur ein: "geht nicht"...

Meine vorletzte VM war eine Devuan, größere Probleme sind mir da auch nicht
bekannt, mich hätten da seine Probleme eben auch interessiert und ggf. wäre damit
ja die Prerequisites für Debian wieder in die Doku gekommen.
Leider hat er ja auch schon bei den ja, seit 40 Jahren bekannten,
einfachen Voraussetzungen in Discussions/Foren:
- ein Problem in eine Discussions
- aussagefähiger Titel in den Eröffnungspost
- genau Dokumentation, wobei das Problem entstanden ist (Versionsnummer, logs etc..)
- ein Problem nach dem anderen
usw...

so seine Probleme...
Dann hat er wohl doch einmal auf ein RTFM! reagiert und da, welch ein Wunder,
dann ging es ja wohl auch mit Debian...

>... Diskussion dazu ist leider zu.

"zu" sind sie dann aber wohl alle, weil er eben
wieder nur "in jeden Raum gekackt hat" --> SPAM...

<del>Ich habe bestimmt mind. vier Stunden in den Typ investiert, aber mit sein
ewigen Geschrei (ist ein Bug! ...ist ein Bug!), hatte auch ich ja schon meine Probleme.
Freetz/freetz.NG, also das Konzept dahinter, menuconfig, wollte und hat er ja
auch nicht verstanden/verstehen wollen, einmal einen Blick in die Doku werfen
war ihm unbekannt...</del>
Ich hätte ihm gerne weiter geholfen, <del>aber er hat ja da noch nicht einmal
meine Links in den Posts angeklickt... Suchmaschinen kennt er wohl auch nicht,
zum (vorläufigen?) Schluß hatte ich ihm schon seine "besondere, coole"
AVM-Firmware gesucht...</del> <-- immer dieses "Gewimmer!", hab's 'mal selbst geändert ;-(

Wie auch immer, ggf. kommen wir ja dann hier zu einer allgemein gültigen
Debian-11-Prerequisites?!
Die dann eben auch noch in die Doku mit rein kommt...

**Reply by kainers1** [reply]
> Wie auch immer, ggf. kommen wir ja dann hier zu einer allgemein gültigen
Debian-11-Prerequisites?!
Die dann eben auch noch in die Doku mit rein kommt...

Ok, dann werde ich meine Erkenntnisse nochmals in einer frischen VM testen, dann kann das in die Doku rein.

**Comment by fda77** [comment]
Ich hab dem guten mal ne woche Urlaub spendiert, kann dauern bis es ne Anwort gibt. Nicht nur dass er massenhaft Diskussionen+Issues mit gleichem Inhalt eröffnete, es gibt auch zeitgleich erstelle Diskussionen+Issues mit gleichem Namen: https://github.com/Freetz-NG/freetz-ng/issues/654 + https://github.com/Freetz-NG/freetz-ng/discussions/657

Was das Problem mit Debian ist weiss ich nicht, man muss halt prerequisites installieren und gut. Das bekommen aber viel nicht hin. Es bräuchte jemanden der das für Anfänger auflistet, am besten hier dazu: https://freetz-ng.github.io/freetz-ng/PREREQUISITES (docs/PREREQUISITES.md)

**Comment by aus-sua** [comment]
> Es bräuchte jemanden der das für Anfänger auflistet...

Das wäre auch mein Vorschlag dazu.
Damit bist Du, fda77, ja erst einmal hier "raus" ;-)

Dann sollten wir das hier eben weiter/wieder durchspielen, hier dokumentieren und dann eben
erst das Ergebnis in die Doku "zurückführen".

Ist es nötig, das bei Debian 10 zu beginnen (meiner Meinung nach: nein), sondern gleich
nur zu Debian 11 beachten/dokumentieren?!

**Reply by fda77** [reply]
Es ist eigentlich egal welche Version, am besten die die man selbst verwendet. Danach kommt eh immer jemand der es für ne andere bräuchte :)

**Reply by pfichtner** [reply]
Siehe dazu https://github.com/Freetz-NG/freetz-ng/discussions/475#discussioncomment-2950468

**Reply by aus-sua** [reply]
Mit 'nem, ggf. da auch immer-zeitnah-nach-gepflegten, bash-script wäre es
ja noch einfacher.
Ok, ich teste das einmal innerhalb dieser Woche mit z.B. devuan-Neuinstallation,
Probleme sehe ich da aber so nicht

PS:
Mir, als alten "Bedenketräger", fällt da aber schon wieder gleich ein:
 zu einfach sollte es man den Leuten eben auch nicht machen...
Diejenigen, die heute schon beim ctr-c/ctr-v scheitern, bekommen dann
auch das url zum script nicht kopiert, um dann aber gleich hier mit "Bug in freetz-NG"
aufzuschlagen... ;-( 
Ein Teufelskreis ;-)

**Comment by CaptainJackDaniels** [comment]
Warum so kompliziert? Weshalb nicht einfach ein Installation check. Denn es ja so schon gibt in freetz-ng.

**Reply by pfichtner** [reply]
Es gibt Systeme, die bereit gestellt werden um Freetz-Images bauen zu können (VMs, Docker-Images,, ...). Überall dort wird/muss die Liste zu installierender Pakete gepflegt werden. Wäre dies maschinenlesbar entral gepflegt, würde dies entfallen. Man könnte daraus sowohl einen automatische Installation als auch eine automatische Überprüfung relalisieren. IMHO prüft der check "nur" ob notwendige Binaries da sind, was bei Anfängern (und nicht nur bei denen) nicht direkt auf zu installierende Pakete rückschliesen lässt.

**Reply by aus-sua** [reply]
Die in den Prerequisites aufgeführten Pakete-Erweiterungen funktionieren ja.
Nur diese pflegt/überwacht auch der Entwickler selbst.
Und damit,
für mich logisch,
endet eben auch die Unterstützung durch den Entwickler genau da!

Das es auch mit anderen Linuxen funktioniert, ist eben ein Bonus.
Nur,
dafür auch die Unterstützung und die dazu nötige Pflege (!) der Versionsverwaltung
der ggf. unterschiedlichen Pakete in freetz-NG einzubauen,
das sollte nicht auch noch die Aufgabe des Entwicklers sein...

**Reply by pfichtner** [reply]
Ich kann Dir nicht ganz folgen. Im Moment wird u.a. https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES.md gepflegt (für unterschiedliche Versionen von Fedora und Ubuntu). 
Mein Vorschlag war lediglich, das Skript aus dem MD-Dokument automatisch zu generieren (oder beies aus der CSV-Datei o.ä.) zu erzeugen. Plfegeaufwand für den Entwickler (btw: wer ist _der_ Entwickler?) ist der gleiche bzw. ändert sich überhaupt nicht.

**Reply by aus-sua** [reply]
Ich Dir auch nicht, zumindest nicht immer.
Ich glaube aber, wir reden nur aneinander vorbei,
weiteres dann morgen Nacht...

**Comment by fda77** [comment]
Ich glaube nicht dass man so ein Script brauchen würde, man muss ja nur diese 1 Zeile kopieren. Falls nötig kann man sie auch gleich anpassen.
Wenn ihr es aber machen  wollt, nur zu. Wäre wie oben schon steht gut wenn das aus docs/ gepart wird, damit nicht an mehreren Stellen was geupdatet werden muss

Aber ich mag Debian wegen diesem ganzen Mist wirklich nicht, @berndy2001 vielleicht kannst du mal nachschauen wie das mit diesen Libs ist und was Debian an eigenes gebastelt hat, oder ob nur irgend was fehlt: https://github.com/Freetz-NG/freetz-ng/issues/671

**Reply by kainers1** [reply]
> Aber ich mag Debian wegen diesem ganzen Mist wirklich nicht

das war mir seit 
> Debian ist dafür bekannt ungeeignet zu sein.

klar.

**Reply by fda77** [reply]
Hab keine Lust auf die ständigen Nachfragen :) Also eigentlich hab ich gar keine Lust auf irgend welchen komischen Fragen. Die könnten auch mal andere beantworten
